### DISCUSSION TITLE
tutorials/intro-hotchocolate/08-rest-datasource

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#84] [course feedback]
# tutorials/intro-hotchocolate/08-rest-datasource

Set up a REST API as a data source using an auto-generated HTTPClient.

https://www.apollographql.com/tutorials/intro-hotchocolate/08-rest-datasource

**Comment by JesseXia** [comment]
I'm running dotnet 8 for the course. And I think the command to generate httpclient has changed. The swagger2csclient doens't exist anymore. Instead, I used `dotnet nswag openapi2csclient /input:Data/swagger.json /classname:SpotifyService /namespace:SpotifyWeb /output:Data/SpotifyService.cs` and it works as expected.

**Reply by mabuyo** [reply]
Thanks for the heads up @JesseXia! The course uses .NET 7 for now, but this is helpful to note for the next iteration 🙏🏽

**Reply by nagababuthota984** [reply]
Remove the extra slashes and run. Below command worked for me.
`dotnet nswag openapi2csclient /input:Data/swagger.json /classname:SpotifyService /namespace:SpotifyWeb /output:Data/SpotifyService.cs`

**Comment by mobeensmb** [comment]
File C:\Users\smbmo\OneDrive\Desktop\React Basics\GraphQLProject\odyssey-intro-hotchocolate\.config\dotnet-tools.json came from another computer and might be blocked to help protect this computer.

I am seeing above issue while running this command. I also tried different ways to unbloc, but no luck. Help here please..... 

dotnet nswag openapi2csclient /input:Data/swagger.json /classname:SpotifyService /namespace:SpotifyWeb /output:Data/SpotifyService.cs

**Reply by mabuyo** [reply]
Hi @mobeensmb ! That's a strange error, since `dotnet-tools.json` isn't in the starter code, it's generated when you run the command `dotnet new tool-manifest`. Can you confirm that the file is newly created when you run `dotnet new tool-manifest`?

**Reply by mabuyo** [reply]
You could also try this solution: https://stackoverflow.com/a/77863931

**Reply by mobeensmb** [reply]
yes, it's newly created while running 'dotnet new tool-manifest' And I am also not seeing unblock option at properties window

**Reply by mabuyo** [reply]
Got it, could you try moving the folder out of OneDrive?? The file permissions with the cloud may be a factor here. Try re-cloning locally or moving the folder out and see if that fixes the issue!

**Comment by opsingh861** [comment]
where can i get swagger.json file?

**Reply by lizhennessy** [reply]
Hi there @opsingh861! The `swagger.json` file should already be included in your project, inside the `Data` directory.

**Reply by opsingh861** [reply]
Got it, Thank you so much!!!

**Comment by mcweglowski** [comment]
I think I may have an interesting case. I pulled the base code for this course from your git repository. The error I am having currently says:
```error CS1061: 'IRequestExecutorBuilder' does not contain a definition for 'RegisterService' and no accessible extension method 'RegisterService' accept 
ing a first argument of type 'IRequestExecutorBuilder' could be found (are you missing a using directive or an assembly ref 
erence?) [C:\Users\mawg\source\repos\github\GraphQLTutorial\odyssey-intro-hotchocolate\Odyssey.MusicMatcher.csproj]```

The error is in the same `Program.cs` file content as here. Below you can see my `csproj` file:
```
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="HotChocolate.AspNetCore" Version="14.0.0" />
    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
  </ItemGroup>
</Project>
```

I am totally puzzled and have no idea what could be wrong.

**Reply by mcweglowski** [reply]
There is something wrong with my markdown. One more time then:
```
error CS1061: 'IRequestExecutorBuilder' does not contain a definition for 'RegisterService' and no accessible extension method 'RegisterService' accept 
ing a first argument of type 'IRequestExecutorBuilder' could be found (are you missing a using directive or an assembly ref 
erence?)
```

csproj:
```
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>net8.0</TargetFramework>
    <Nullable>enable</Nullable>
    <ImplicitUsings>enable</ImplicitUsings>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="HotChocolate.AspNetCore" Version="14.0.0" />
    <PackageReference Include="Newtonsoft.Json" Version="13.0.3" />
  </ItemGroup>

</Project>

```

**Reply by lizhennessy** [reply]
hey there @mcweglowski ! At what point in the lesson did you see this error pop up? From what I can tell your `.csproj` file looks good, but I'm wondering if something went wrong when adding certain dependencies during this lesson.

**Reply by scottrudy** [reply]
I hit this same thing and for the same reason. The tutorial explicitly uses v13 of the library and your project shows v14, which no longer supports/needs that method as documented in the [breaking changes](https://chillicream.com/docs/hotchocolate/v14/migrating/migrate-from-13-to-14).

TLDR; You don't need to register the service with the GraphQL Server if using HotChocolate.AspNetCore >= 14.0

**Reply by lizhennessy** [reply]
Hi @scottrudy , thanks for calling this out. Our update with version 14 is well underway and should be live in the new year - hopefully this will take care of the mismatch for future learners!

**Comment by HobDev** [comment]
I am using PostgreSQL on Azure as data source for GraphQL API's in my app. So I want to learn about that. Upto this point all the content was very helpful. But now I am stuck because of different data source

**Reply by mabuyo** [reply]
Hey there @HobDev! Sorry to hear that you feel stuck. I encourage you to check out the Hot Chocolate documentation on ["Fetching from databases"](https://chillicream.com/docs/hotchocolate/v15/fetching-data/fetching-from-databases) to start. It looks like they [have a few integrations](https://chillicream.com/docs/hotchocolate/v15/integrations), but unfortunately not PostgreSQL.
